File: warehouse-realtime/src/main/scala/com/lavideo/util/MyKafkaUtil.scala
package com.lavideo.util

import java.util.Properties

import org.apache.kafka.clients.consumer.ConsumerRecord
import org.apache.kafka.common.serialization.StringDeserializer
import org.apache.spark.streaming.StreamingContext
import org.apache.spark.streaming.dstream.InputDStream
import org.apache.spark.streaming.kafka010.{ConsumerStrategies, KafkaUtils, LocationStrategies}

/**
 * 连接kafka工具类
 */
object MyKafkaUtil {
    //创建配置信息对象
    private val properties: Properties = PropertiesUtil.load("config.properties")
    //初始化连接到集群的地址
    private val broker_List: String = properties.getProperty("kafka.broker.list")

    //kafka消费者配置
    private val kafkaPara: Map[String, Object] = Map(
        "bootstrap.servers" -> broker_List,
        "key.deserializer" -> classOf[StringDeserializer],
        "value.deserializer" -> classOf[StringDeserializer],
        "group.id" -> "lavkafka",
        "auto.offset.reset" -> "latest",
        "enable.auto.commit" -> (true: java.lang.Boolean)
    )

    //创建DStream kafka数据流返回数据
    def getKafkaStream(ssc : StreamingContext,topic : Set[String]) : InputDStream[ConsumerRecord[String,String]] = {
        val dStream : InputDStream[ConsumerRecord[String, String]] = KafkaUtils.createDirectStream[String,String](
            ssc,
            LocationStrategies.PreferConsistent,
            ConsumerStrategies.Subscribe[String, String](topic, kafkaPara)
        )
        dStream
    }
}

File: datawarehouse-realtime/src/main/java/com/lavideo/utils/MyKafkaUtil.java
package com.lavideo.utils;


/**
 *
 */

import org.apache.flink.api.common.serialization.SimpleStringSchema;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.DataStreamSource;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;


import org.apache.flink.streaming.connectors.kafka.FlinkKafkaConsumer011;

import java.util.Properties;

public class MyKafkaUtil{

    public static DataStream<String> getKafkaStream(StreamExecutionEnvironment env, String topic, String bootstrapServers, String groupId) {
        Properties prop = new Properties();
        prop.setProperty("bootstrap.servers",bootstrapServers);
        prop.setProperty("group.id",groupId);
        prop.setProperty("key.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        prop.setProperty("value.deserializer", "org.apache.kafka.common.serialization.StringDeserializer");
        prop.setProperty("auto.offset.reset", "latest");

        DataStreamSource<String> kafkaStream = env.addSource(
                new FlinkKafkaConsumer011<String>(topic, new SimpleStringSchema(), prop)
        );
        return kafkaStream;

    }


}
